test/src/test/java/org/corfudb/integration/StreamAddressDiscoveryIT.java

```diff
@@ -532,13 +533,13 @@ public void checkpointAndTrimAtDifferentPoint() throws Exception {
             final int sizeMap1 = 4;
             final int sizeMap2 = 6;
 
-            Map<String, String> map1 = defaultRT.getObjectsView().build()
+            StreamingMap<String, String> map1 = defaultRT.getObjectsView().build()
```

Comment:
these will involve about 168 changes in upstream since Map<> is used in those places.

Reply:
There is nothing forbidding you from using it as a Map. The only place where this change will be needed is wherever MultiCheckpointWriter is being consumed.

Reply:
In other words, the change in the contract happens here:

`public class MultiCheckpointWriter<T extends StreamingMap> `

Reply:
I checked the corfu-compactor logic, we consume `MultiCheckpointWriter` using `ICorfuMap`:

`MultiCheckpointWriter<ICorfuMap> mcWriter = new MultiCheckpointWriter<>();`

And `ICorfuMap` is defined as:

```
public interface ICorfuMap<K, V>
    extends StreamingMap<K, V> {
```

Thus, from what I can tell no changes should be necessary on the client side.

Reply:
MultiCheckpointWriter shouldn't even use generics since we never need them.

I think its more appropriate to introduce new tests to test `StreamingMap` and not change existing ones.

Reply:
You technically don't need the generic parameter. In this case it serves more as a type alias.

The tests have not been changed, just the upper bound type associated with the constructed objects.